### ISSUE TITLE
OpenStack provider support and templates

### ISSUE THREAD

**Issue by DinaBelova** [k0rdent/kcm#591]
**Goals**
- Support OpenStack as a target infrastructure provider

**Major deliverables**
- Create two OpenStack templates as we have them for AWS just deployed into OpenStack, with same relevant values as they already exist for the AWS templates ([aws-hosted-cp](https://github.com/Mirantis/hmc/blob/main/templates/aws-hosted-cp/values.yaml) and [aws-standalone-cp](https://github.com/Mirantis/hmc/blob/main/templates/aws-standalone-cp/values.yaml))

**Who it benefits**
- Customer: Less work getting started with project on OpenStack
- Mirantis: Possibility to enable customers wanting to deploy into OpenStack

**Acceptance criteria**
- openstack-standalone-cp and openstack-hosted-cp templates existing
with equivalent variables as the AWS templates [aws-hosted-cp](https://github.com/Mirantis/hmc/blob/main/templates/aws-hosted-cp/values.yaml) and [aws-standalone-cp](https://github.com/Mirantis/hmc/blob/main/templates/aws-standalone-cp/values.yaml)
- Automated testing
- OpenStack Provider support

**Assumptions**
- OpenStack CAPI Provider (CAPO) is without bugs or k0s/k0smotron related issues

**Telemetry & Success Criteria**
- As with OpenStack templates we need a telemetry for: 
    - each cluster that is created with the template
    - heartbeat information from each cluster that is running

**Limitations**
- none

**Comment by zerospiel** [comment]
@bnallapeta could you confirm the current relevance of the sub-tasks, are they still opened or some of them had already been done? for instance, I can see the OS docs https://docs.k0rdent.io/v0.1.0/template-openstack/ so I am assuming #595 has been done (not sure though about the #593)

**Comment by bnallapeta** [comment]
@zerospiel Of the 3 tasks that are open, 
- Hosted CP is currently blocked due to an upstream issue.
- e2e tests are pending.
- Have asked for clarification on 595.

**Comment by zerospiel** [comment]
@bnallapeta could you give updates regarding the 593 and 596 issues? is the status still the same?

**Comment by bnallapeta** [comment]
@zerospiel yes, its still the same.

**Comment by michaelschencksap** [comment]
I'm attempting to leverage the credential distribution process with openstack, and can't tell if it's just not yet supported.  I was under the impression it already is supported based on [the documentation](https://docs.k0rdent.io/next/admin/access/credentials/credentials-propogation/#openstack) however any cluster deployments to namespaces other thank `kcm-system` result in capo claiming `"secrets \"openstack-cloud-config\" not found"`

I do see that `AccessManagement` is working because it is creating the `Credentials` in the other namespaces.  When I create a ClusterDeployment in `kcm-system` it works.

Should this be working or is this still in development?

**Comment by zerospiel** [comment]
@michaelschencksap hey, might the problem be related to #943 issue? Because it seems that the `cld` requires everything to be within a namespace whereas in `CAPO` it is implemented in another way

**Comment by bnallapeta** [comment]
@michaelschencksap Yes, as @zerospiel pointed out, atm, OpenStack only supports secrets in the same namespace. And the reason for that is the non-existence of something like [OpenStackClusterIdentity](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/2386) which I have raised and discussed upstream. So until a new CR like that is supported by CAPO, we will be limited to deploying the secret in the kcm-system namespace itself.
